src/NHibernate/Mapping/Table.cs

```diff
@@ -1035,7 +1036,10 @@ public IEnumerable<string> ValidateColumns(Dialect.Dialect dialect, IMapping map
 				}
 
 				//TODO: Add new method to ColumnMetadata :getTypeCode
-				bool typesMatch = column.GetSqlType(dialect, mapping).StartsWith(columnInfo.TypeName, StringComparison.OrdinalIgnoreCase);
```

Comment:
Why not simply check that next char is not a letter (`char.IsLetter`)?

Reply:
I was/am thinking about that. Even have code about to push. But it is not that simple.

Reply:
Still see no reasons to use regex - check additionally for not a digit (`char.IsDigit`). Alphanumeric character check should suffice

Reply:
Yes, I hate the regex myself. But that is not that easy really. For example, Db2's `CHAR(16) FOR BIT DATA` would match to `CHAR(16)` even in my implementation with regex.